mlflow/tracking/fluent.py

```diff
@@ -988,6 +989,100 @@ def log_image(image: Union["numpy.ndarray", "PIL.Image.Image"], artifact_file: s
 
 
+def log_table(
+    data: Union[Dict[str, Any], pd.DataFrame],
+    local_path: Optional[str] = None,
+) -> None:
+    """
+    Log a table to MLflow artifact store.
+
+    :param data: Dictionary or pandas.DataFrame to log.
+    :param local_path: Path to the file to write.
+    :return: None
+
+    .. test-code-block:: python
+        :caption: Dictionary Example
+
+        import mlflow
+
+        table_dict = {
+            "inputs": ["What is MLflow?", "What is Databricks?"],
+            "outputs": ["MLflow is ...", "Databricks is ..."],
+            "toxicity": [0.0, 0.0],
+        }
+
+        with mlflow.start_run():
+            # Log the dictionary as a table
+            mlflow.log_table(data=table_dict, local_path="qabot_eval_results.csv")
+
+    .. test-code-block:: python
+        :caption: Pandas DF Example
+
+        import mlflow
+        import pandas as pd
+
+        table_dict = {
+            "inputs": ["What is MLflow?", "What is Databricks?"],
+            "outputs": ["MLflow is ...", "Databricks is ..."],
+            "toxicity": [0.0, 0.0],
+        }
+        df: pd.DataFrame = pd.DataFrame.from_dict(table_dict)
+
+        with mlflow.start_run():
+            # Log the df as a table
+            mlflow.log_table(data=df, local_path="qabot_eval_results.csv")
+    """
+    if isinstance(data, Dict):
+        data = pd.DataFrame(data)
+
+    if not isinstance(data, pd.DataFrame):
+        raise ValueError("data must be a pandas.DataFrame or a dictionary")
+
+    run = _get_or_start_run()
+    run_id = run.info.run_id
+    run_artifact_path = None
+    LLM_ARTIFACT_NAME = local_path or "llm_predictions.csv"
+    TAG_NAME = "mlflow.loggedTables"
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        artifacts = [f.path for f in client.list_artifacts(run_id)]
+        if LLM_ARTIFACT_NAME in artifacts:
+            run_artifact_path = mlflow.artifacts.download_artifacts(
+                run_id=run_id, artifact_path=LLM_ARTIFACT_NAME, dst_path=tmpdir
+            )
+            _logger.info(
+                "Appending new inputs to already existing artifact "
+                f"{LLM_ARTIFACT_NAME} for run {run_id}."
+            )
+        else:
+            run_artifact_path = os.path.join(tmpdir, LLM_ARTIFACT_NAME)
+            _logger.info(f"Creating a new {LLM_ARTIFACT_NAME} for run {run_id}.")
+
+    if os.path.exists(run_artifact_path):
+        existing_predictions = pd.read_csv(run_artifact_path)
+        if existing_predictions.shape[0] + data.shape[0] > 1000:
+            _logger.warning(
+                f"Trying to log a {LLM_ARTIFACT_NAME} with length "
+                "more than 1000 records. It might slow down performance."
+            )
+            # combine the existing predictions with the new predictions
+            combined_predictions = pd.concat([existing_predictions, data], ignore_index=True)
+            combined_predictions.to_csv(run_artifact_path)
```

Comment:
```suggestion
            combined_predictions.to_csv(run_artifact_path, index=False)
```

Can we drop `index`?